moodle/moodle.go

```diff
@@ -410,7 +412,18 @@ func (c *Controller) createDeployment(foo *operatorv1.Moodle) (error, string, st
 	mysqlUserName := foo.Spec.MySQLUserName
 	fmt.Printf("MySQL Username:%v\n", mysqlUserName)
 
-	mysqlUserPassword := foo.Spec.MySQLUserPassword
+	secretsClient := c.kubeclientset.CoreV1().Secrets(namespace)
+	secret, err := secretsClient.Get(foo.Spec.MySQLPassword, metav1.GetOptions{})
+
+	if err != nil {
+		fmt.Printf("Error, secret %s, not found from in namespace %s: %v\n", foo.Spec.MySQLPassword, namespace, err)
+	}
+	if _, ok := secret.Data["mySQLPassword"]; !ok {
```

Comment:
The logic here needs to be:

- Parse secretName and keyName from foo.Spec.mySQLUserPassword
  - The value will have following structure: <secretName>-<key-name>
- Get the Secret Object using secretName
- Get the secret.Data[keyName]
- Decode the data